### PR TITLE
feat(inputs.prometheus): use namespace annotations to filter pods to be scraped

### LINKED ISSUE #12776: Allow input.prometheus to select pods based on namespace annotation
### Use Case

It is quite common for Kubernetes namespace to store flags/settings in form of annotations. One example is whether namespace is a production or dev environment.  It would be nice if input.prometheus can filter pods based on namespace annotations.

### Expected behavior

Config option similar to `tagpass`/`tagdrop`, but for namespace annotation. If namespace tags match `annotation_pass` then pods in namespace are further considered for scraping, subject to other plugin settings. Likewise for `annotation_drop`: when matches, pods wont be scraped.

### Actual behavior

-

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by redbaron**
- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #12776

### CODE REVIEW

File: plugins/inputs/prometheus/sample.conf [hunk 1]
```diff
@@ -125,3 +125,13 @@
 
   ## Use TLS but skip chain & host verification
   # insecure_skip_verify = false
+
+  ## Control pod scraping based on pod namespace annotations
+  #
+  # Pass and drop here act like tagpass and tagdrop, but instead
+  # of filtering metrics they filters pod candidates for scraping
```
Comment (powersj) [inline plugins/inputs/prometheus/sample.conf:132]:
```suggestion
  ## Control pod scraping based on pod namespace annotations
  ## Pass and drop here act like tagpass and tagdrop, but instead
  ## of filtering metrics they filters pod candidates for scraping
```

File: plugins/inputs/prometheus/sample.conf [hunk 2]
```diff
@@ -125,3 +125,13 @@
...
+  #[inputs.prometheus.namespace_annotation_pass]
+  # annotation_key = ["value1", "value2"]
+  #[inputs.prometheus.namespace_annotation_drop]
+  #
```
Comment (powersj) [inline plugins/inputs/prometheus/sample.conf:136]: While the default value for both of these is not set, can you add an example like to the annotation_drop like you have for pass?

**Review by powersj — COMMENTED** [review]
Thanks for the PR! Looks good, thanks for the clean up as well. Two comments on the sample conf + `make docs` afterwards

File: plugins/inputs/prometheus/sample.conf (hunk 2 above)
Comment (redbaron) [reply plugins/inputs/prometheus/sample.conf]: done

**Review by powersj — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
LGTM. Thanks for your contribution @redbaron!

**Comment by srebhan** [comment]
@redbaron can you please fix the merge conflict!?

**Comment by redbaron** [comment]
@srebhan , done

**Review by srebhan — APPROVED** [review]
Thanks @redbaron!

**Pull request merged**